qutebrowser/browser/webengine/webview.py

```diff
@@ -144,6 +146,18 @@ def _set_bg_color(self):
             col = self._theme_color
         self.setBackgroundColor(col)
 
+    @pyqtSlot(QUrl, 'QAuthenticator*', 'QString')
+    def _on_proxy_authentication_required(self, url, authenticator, proxyHost):
+        answer = shared.authentication_required(url, authenticator, [self.shutting_down, self.loadStarted])
```

Comment:
That seems like a quite confusing message, because it's not the page itself which requires authentication. I think it'd be better to show "Proxy authentication required" as title, and something like "<url> requires authentication for <proxy-url>" or so as text.

Also see the related QtWebKit code:

https://github.com/qutebrowser/qutebrowser/blob/66e4c3286a0ba010214c1acc20193fbecc63f376/qutebrowser/browser/webkit/network/networkmanager.py#L305-L314

With the difference that we know `url` but don't have a realm anymore (which is why it probably doesn't make sense to add something to `shared`).

Reply:
Sorry, didn't see that webkit/networkmanager.py also has a `on_proxy_authentication_required()`. I'll update the WebEngine one to more closely follow the WebKit one.